receiver/otlpreceiver/config.go

```diff
@@ -71,24 +71,13 @@ func (cfg *Config) Unmarshal(componentParser *configparser.Parser) error {
 	// next manually search for protocols in viper, if a protocol is not present it means it is disable.
 	protocols := cast.ToStringMap(componentParser.Get(protocolsFieldName))
 
-	// UnmarshalExact will ignore empty entries like a protocol with no values, so if a typo happened
-	// in the protocol that is intended to be enabled will not be enabled. So check if the protocols
-	// include only known protocols.
-	knownProtocols := 0
 	if _, ok := protocols[protoGRPC]; !ok {
 		cfg.GRPC = nil
-	} else {
-		knownProtocols++
 	}
 
 	if _, ok := protocols[protoHTTP]; !ok {
 		cfg.HTTP = nil
-	} else {
-		knownProtocols++
 	}
 
-	if len(protocols) != knownProtocols {
-		return fmt.Errorf("unknown protocols in the OTLP receiver")
-	}
 	return nil
 }
```

Comment:
This becomes easier now with `Exists` I think. 

```
protocols := componentParser.Sub(protocolsFieldName)
if !protocols.Exists(protoGRPC) {
	cfg.GRPC = nil
}
...
```

I think this probably can be done as a separate PR anyway :)) Sorry to ask you to clean something else, but while reading I found this code too complicated.

Reply:
Not a problem, and I've updated since it's better to use the parser directly and I'm changing the flow here anyway.